File: spark-demo/src/main/scala/WordCount.scala
import java.util
import java.util.{Arrays, List}

import org.apache.spark.api.java.{JavaPairRDD, JavaRDD}
import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SparkSession

/**
  * Created by DebugSy on 2017/8/22.
  */
object WordCount {

  def main(args: Array[String]): Unit = {
    val spark = SparkSession.builder.appName("JavaWordCount").master("local").getOrCreate

    val lines: RDD[String] = spark.read.textFile(args(0)).rdd

    val words: RDD[String] = lines.flatMap(line => line.split(" "))

    val pairs = words.map(word => (word, 1))

    val wordCounts = pairs.reduceByKey(_+_)

    wordCounts.foreach(wordNumberPair => println(wordNumberPair._1 + " : " + wordNumberPair._2))

    spark.stop()

  }

}

File: spark-demo/src/main/java/com/shiy/spark/WordCount.java
package com.shiy.spark;

import org.apache.commons.lang.ArrayUtils;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.function.FlatMapFunction;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.SparkSession;

import java.util.Arrays;
import java.util.Iterator;
import java.util.regex.Pattern;

/**
 * Created by DebugSy on 2017/8/24.
 */
public class WordCount {

	private static final Pattern SPACE = Pattern.compile(" ");

	public static void main(String[] args) {

		SparkSession sparkSession = new SparkSession.Builder().appName("word count").master("local").getOrCreate();

		JavaRDD<String> lines = sparkSession.read().textFile("E:/tmp/data.txt").toJavaRDD();

		JavaRDD<String> words = lines.flatMap(s -> Arrays.asList(SPACE.split(s)).iterator());

//		words.mapToPair()

	}

}
